runners/spark/src/main/java/org/apache/beam/runners/spark/stateful/StateSpecFunctions.java

```diff
@@ -177,8 +185,14 @@
 
           // close and checkpoint reader.
           reader.close();
-          LOG.info("Source id {} spent {} msec on reading.", microbatchSource.getId(),
-              stopwatch.stop().elapsed(TimeUnit.MILLISECONDS));
+          final long readDurationMillis = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
+
+          readDuration.set(readDurationMillis);
```

Comment:
I agree, while moving this to `ReadReportDStream#compute` will not entirely solve this, as the metrics reporter till samples the value once in a reporting-period, it does open the door to lowering the reporting-period so that it actually reports the value (instead of sampling).